### PR TITLE
Support for DnD5e 3.x

### PR DESCRIPTION
**Pull request by BeardedGnome**
First draft at updating Token-Mold to support DnD5e 3.x and Foundry v11.

### CODE REVIEW

File: scripts/token-mold.js [hunk 1]
```diff
@@ -1285,8 +1278,11 @@ class TokenMoldForm extends FormApplication {
     };
 
     let barAttributes = [];
-    for (let type in game.system.model.Actor) {
-      let newAttributes = getAttributes(game.system.model.Actor[type]).map(
+
+    const types = Actor.implementation.TYPES;
+    const shells = types.map(t => new Actor.implementation({ name: t, type: t }));
+    for (const actorElement of shells) {
+      let newAttributes = getAttributes(actorElement.toObject().system).map(
```
Comment (Lochspring) [inline scripts/token-mold.js:1285]: Do we need toObject() here?  What type is actorElement? A JSON string?

Also, you can get away with const here.  newAttributes isn't changing in this loop once established, and it'll get overwritten for the next iteration anyway (consts are scoped to the block or iteration).

File: scripts/token-mold.js [hunk 2]
```diff
@@ -1317,11 +1313,12 @@ class TokenMoldForm extends FormApplication {
...
       let sortFun = function (a, b) {
-        if (a.attribute > b.attribute) return 1;
-        else if (a.attribute < b.attribute) return -1;
+        if (a.attribute > b.attribute) { return 1; }
+        else if (a.attribute < b.attribute) { return -1; }
         return 0;
       };
-      for (let skill of Object.keys(game.system.model.Actor["npc"].skills)) {
+      const npc = shells.find(o => o.type === 'npc');
+      for (let skill of Object.keys(npc.toObject().system.skills)) {
```
Comment (Lochspring) [inline scripts/token-mold.js:1321]: same here:  do we need toObject()?  Kind of comes back to what is the data type of `shells`?

**Review by Lochspring — COMMENTED** [review]
Minor questions.  Ignoring formatting.  INstall Prettier and ESlint into your IDE!

File: scripts/token-mold.js (hunk 2 above)
Comment (BeardedGnome) [reply scripts/token-mold.js:1441]: The `toObject()` was copied from the example code here: https://foundryvtt.com/article/active-effects/

I assume it's required (or optimal).

**Comment by BeardedGnome** [comment]
> Minor questions. Ignoring formatting. INstall Prettier and ESlint into your IDE!

I did. They helped a little. And made the PR bigger.

File: scripts/token-mold.js (hunk 1 above)
Comment (BeardedGnome) [reply scripts/token-mold.js]: Modified this to create `map[type] = Actor`. Iteration is the same, but the 'npc' look up later on is faster.

The `toObject()` was copied from the example code here: https://foundryvtt.com/article/active-effects/

I assume it's required (or optimal).

File: scripts/token-mold.js [hunk 3]
```diff
@@ -1285,30 +1391,37 @@ class TokenMoldForm extends FormApplication {
...
-      let newAttributes = getAttributes(game.system.model.Actor[type]).map(
-        (e) => e.join(".")
+
+    const types = Actor.implementation.TYPES;
+    const shellMap = new Map(
+      types.map((t) => [t, new Actor.implementation({ name: t, type: t })]),
+    );
+    shellMap.forEach((value, key, map) => {
+      const newAttributes = getAttributes(value.toObject().system).map((e) =>
+        e.join("."),
```
Comment (BeardedGnome) [inline scripts/token-mold.js:1401]: New `map` code.

Comment (Lochspring) [reply scripts/token-mold.js:1401]: I'm still not sure you need the toObject here:  the props of `npc.system` stilll contain what you want to pull out, just more of it.  However, the toObject() call strips off all the primitives and additional functions and model data from CreatureTemplate, which makes the object leaner.  Effectively, you're extracting all of the iterable props from the npc.system object to re-configure just the parts needed.  no objection;  there's not enough of a performance hit where we care.

**Review by Lochspring — APPROVED** [review]
I'm happy with the changes here.  We can talk bug fixes if you like, but the new map code should be a little faster than the old approach.

**Pull request merged**